AddUtil.java

```diff
@@ -16,10 +16,60 @@ public static void myassert(boolean  x) {
     }
     private static int add(int a, int b){
         //Write your code here
+        return (a + b);
     }
 
     private static String add(String a, String b) {
         //Write your code here
+        if((a.charAt(0) == '-') && (b.charAt(0) == '-')) {
+            return "-" + addPvt(a.substring(1), b.substring(1));
+        }
+        else {
+            return addPvt(a, b);
+        }
+    }
+
+        private static String addPvt(String a, String b) {
+        // Handle some base cases
+        if(a.length() == 0){
+            return b;
+        }
+        if(b.length() == 0){
+            return a;
+        }
+        StringBuilder sum = new StringBuilder();
+        int aLen = a.length()-1, bLen = b.length()-1;
+        int tempSum = 0, remainder = 0;
+        try {
+            while ((aLen >= 0) && (bLen >= 0)) {
+                tempSum = (a.charAt(aLen) - '0') + (b.charAt(bLen) - '0') + remainder;
+                remainder = tempSum / 10;
+                tempSum = tempSum % 10;
+                sum.append(tempSum);
+                aLen--;
+                bLen--;
+            }
+            while(aLen >= 0){
+                tempSum = (a.charAt(aLen) - '0') + remainder;
+                sum.append(tempSum);
+                remainder = 0;
+                aLen--;
+            }
+            while(bLen >= 0){
```

Comment:
ya good point, but wondering how I can do it here? any inputs is appreciated.. if i have to write a reusable function, then i need to pass lot of parameters..